test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs

```diff
@@ -209,5 +260,102 @@ internal static void GetDirectoryPaths (out string rootSourceDirectory, out stri
+			test.TestState.AnalyzerConfigFiles.Add (
+						("/.editorconfig", SourceText.From (@$"
+is_global = true
+build_property.{analyzerMSBuildPropertyOption} = true"
+)));
+			if (numberOfIterations != null) {
+				test.NumberOfIncrementalIterations = numberOfIterations;
+				test.NumberOfFixAllIterations = numberOfIterations;
+			}
+			test.FixedState.ExpectedDiagnostics.AddRange (fixedExpected);
+			return test.RunAsync ();
+		}
+
+		internal static Task VerifyDiagnostic<TAnalyzer> (
+			string source,
+			DiagnosticResult[] baselineExpected)
+			where TAnalyzer : DiagnosticAnalyzer, new() =>
+			CSharpAnalyzerVerifier<TAnalyzer>.VerifyAnalyzerAsync (
+				source,
+				TestCaseUtils.UseMSBuildProperties (GetAnalyzerMSBuildPropertyOption<TAnalyzer> ()),
+				baselineExpected);
+
+		internal static DiagnosticResult GetDiagnosticResult<TAnalyzer> ()
+			where TAnalyzer : DiagnosticAnalyzer, new()
+		{
+			return CSharpAnalyzerVerifier<TAnalyzer>.Diagnostic (GetDiagnosticId<TAnalyzer> ());
+		}
+
+		private static Dictionary<Type, string> diagnosticDict = new Dictionary<Type, string>
+		{
+			{typeof(RequiresUnreferencedCodeAnalyzer), RequiresUnreferencedCodeAnalyzer.DiagnosticId},
+			{typeof(RequiresAssemblyFilesAnalyzer), RequiresAssemblyFilesAnalyzer.IL3002}
+		};
+
+		private static string GetDiagnosticId<TAnalyzer> () where TAnalyzer : DiagnosticAnalyzer
+		{
+			if (diagnosticDict.ContainsKey (typeof (TAnalyzer))) {
+				return diagnosticDict[typeof (TAnalyzer)];
+			} else {
+				throw new ArgumentException ($"Couldn't retrieve diagnostic id data for unrecognized Analyzer {typeof (TAnalyzer).Name}");
+			}
+		}
+
+		private static Dictionary<Type, string> attributeDefinitionDict = new Dictionary<Type, string>
+		{
+			{typeof(RequiresUnreferencedCodeAnalyzer), requiresUnreferencedCodeAttributeDefinition},
+			{typeof(RequiresUnreferencedCodeCodeFixProvider), requiresUnreferencedCodeAttributeDefinition},
+			{typeof(RequiresAssemblyFilesAnalyzer), requiresAssemblyFilesAttributeDefinition},
+			{typeof(UnconditionalSuppressMessageCodeFixProvider), unconditionalSuppressMessageAttributeDefinition}
+		};
+
+		private static string GetAttributeDefinition<T> ()
+		{
+			if (attributeDefinitionDict.ContainsKey (typeof (T))) {
+				return attributeDefinitionDict[typeof (T)];
+			} else {
+				throw new ArgumentException ($"Couldn't retrieve attribute information for unrecognized type {typeof (T).Name}");
+			}
+		}
+
+		private static Dictionary<Type, string> msbuildOptionsDict = new Dictionary<Type, string>
+		{
+			{typeof(RequiresUnreferencedCodeAnalyzer), MSBuildPropertyOptionNames.EnableTrimAnalyzer},
+			{typeof(RequiresAssemblyFilesAnalyzer), MSBuildPropertyOptionNames.EnableSingleFileAnalyzer}
+		};
+
+		private static string GetAnalyzerMSBuildPropertyOption<TAnalyzer> () where TAnalyzer : DiagnosticAnalyzer
+		{
+			if (msbuildOptionsDict.ContainsKey (typeof (TAnalyzer))) {
+				return msbuildOptionsDict[typeof (TAnalyzer)];
```

Comment:
This is usually the wrong way to use a `Dictionary`. If you want try to retrieve an item you should almost always use

```C#
if (msbuildOptionsDict.TryGetValue(key, out var value)) {
   return value;
}
...
```

instead of doing a double lookup.